File: exercism/javascript/rna-transcription/rna-transcription.js
//
// This is only a SKELETON file for the 'RNA Transcription' exercise. It's been provided as a
// convenience to get you started writing code faster.
//


const DNA_TO_RNA_MAPPING = new Map(Object.entries({
  'G': 'C',
  'C': 'G',
  'T': 'A',
  'A': 'U'
}));

export const toRna = (dna) => {
  return dna.split('').map(nucleotide => DNA_TO_RNA_MAPPING.get(nucleotide)).join('');
};

File: exercism/ocaml/rna-transcription/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let to_rna l =
    List.map (fun x -> match x with
    | `G -> `C
    | `C -> `G
    | `T -> `A
    | `A -> `U
    ) l
